python/ray/_private/node.py

```diff
@@ -177,9 +178,34 @@ def __init__(
 
         # Register the temp dir.
         if head:
-            # date including microsecond
-            date_str = datetime.datetime.today().strftime("%Y-%m-%d_%H-%M-%S_%f")
-            self._session_name = f"session_{date_str}_{os.getpid()}"
+            maybe_key = None
+            if self._ray_params.external_addresses is not None:
+                self._redis_address = self._ray_params.external_addresses[0]
+                parts = self._redis_address.split("://", 1)
+                enable_redis_ssl = False
+                if len(parts) == 1:
+                    redis_ip_address, redis_port = parts[0].rsplit(":", 1)
+                else:
+                    if len(parts) != 2 or parts[0] not in ("redis", "rediss"):
+                        raise ValueError(f"Invalid redis address {self._redis_address}")
```

Comment:
Can you improve the error message to include what's the definition of "valid redis address"?

Reply:
It'd be great to follwo https://spark.apache.org/error-message-guidelines.html

Reply:
Elaborated. This part comes from the `start_gcs_server` logic in `services.py` though

Reply:
many of our error messages are bad lol... We even had a project to "write better error message" in the past haha..